R/assay-data.R

```diff
@@ -1,129 +1,82 @@
-  ## this when the previous dataset has sample filters entered. This acts
-  ## as a defense to that, and also works to handle this strange case from the
-  ## backend side, too -- perhaps a user will stumble on this in their analyses?
-  if (is.null(samples) || (is.data.frame(samples) && nrow(samples) == 0L)) {
-    samples <- fetch_samples(x)
-  }
-  assert_sample_subset(samples)
-
-  assays <- unique(features$assay)
-  n.assays <- length(assays)
-  if (n.assays > 1L && as.matrix) {
-    stop("Fetching from multiple assays requires return in melted form")
-  }
-
-  if (!identical(aggregate.by, "none")) {
-    stopifnot(n.assays == 1L)
-    if (!normalized) {
-      warning("You probably don't want to aggregate.by on unnormalized data",
+    if (verbose) {
+      warning("No normalization procedure for ", atype, " assay",
               immediate.=TRUE)
     }
+    out <- vals
   }
-
-  out <- lapply(assays, function(a) {
-    f <- filter(features, assay == a)
-    .fetch_assay_data(x, a, f$feature_id, samples, normalized, as.matrix,
-                      subset.threshold, aggregate.by, verbose=verbose)
-  })
-
-  if (length(out) == 1L) {
-    out <- out[[1L]]
-  } else if (!as.matrix) {
-    ## We stop if we are asking for a matrix across multiple assays, but maybe
-    ## we don't have to ... (in the future, I mean)
-    out <- bind_rows(out)
-  }
-
   out
 }
 
-.fetch_assay_data <- function(x, assay_name, feature_ids, samples,
-                              normalized=FALSE, as.matrix=FALSE,
-                              subset.threshold=700, aggregate.by=c("none", "ewm", "zscore"),
-                              ..., verbose=FALSE) {
-  #  stopifnot(is.FacileDataSet(x))
+#' @importFrom checkmate assert_string assert_character assert_flag assert_number
+#' @importFrom rhdf5 h5read
+fetch_assay_data_tbl.FacileDataSet <- function(x,
+                              assay_name,
+                              feature_ids,
+                              samples,
+                              normalized = FALSE,
+                              as.matrix = FALSE,
+                              subset.threshold = 700,
+                              aggregate.by = c("none", "ewm", "zscore"),
+                              verbose = FALSE,
+                              ...) {
+  
   aggregate.by = match.arg(aggregate.by)
   assert_string(assay_name)
   assert_character(feature_ids, min.len=1L)
   samples <- assert_sample_subset(samples)
   assert_flag(normalized)
   assert_flag(as.matrix)
   assert_number(subset.threshold)
-
+  
   finfo <- assay_feature_info(x, assay_name, feature_ids=feature_ids) %>%
     collect(n=Inf) %>%
     arrange(hdf5_index)
   atype <- finfo$assay_type[1L]
   ftype <- finfo$feature_type[1L]
   sinfo <- assay_sample_info(x, assay_name, samples) %>%
     mutate(samid=paste(dataset, sample_id, sep="__"))
-  bad.samples <- is.na(sinfo$hdf5_index)
+  bad.samples <- is.na(pull(sinfo, hdf5_index))
```

Comment:
Add comment for this 'pull'

Reply:
Hah! I learned a new `dplyr` verb today, nice thanks! :-)